server/tests/rest/service_shift/test_service_shift.py

```diff
@@ -98,5 +98,40 @@ def test_get_service_shift(self, mock_list_use_case):
         self.assertEqual(parsed_objects, expected_shifts)
         mock_list_use_case.assert_called_once()
 
+    @patch("application.rest.service_shifts.service_shifts_list_use_case")
+    def test_get_service_shift_with_shelter_id_filter(self, mock_list_use_case):
+        # Arrange: Define the expected shift for a specific shelter
+        test_shelter_id = "ID1"
+        expected_shift = {
+          "_id": 301,
+          "shelter_id": test_shelter_id,
+          "shift_start": 10,
+          "shift_end": 20,
+          "required_volunteer_count": 1,
+          "max_volunteer_count": 5,
+          "can_sign_up": True,
+          "shift_name": "Default Shift"
+        }
+        mock_list_use_case.return_value = [expected_shift]
+
+        # Act: Send GET request with shelter_id filter
+        response = self.client.get(
+            f"/service_shift?shelter_id={test_shelter_id}")
```

Comment:
the GET /service_shift API endpoint still converts shelter_id that's passed to an int, which is probably why this test case fails.